v2/pkg/protocols/headless/engine/http_client.go

```diff
@@ -53,14 +52,13 @@ func newHttpClient(options *types.Options) (*http.Client, error) {
 			transport.Proxy = http.ProxyURL(proxyURL)
 		}
 	} else if types.ProxySocksURL != "" {
-		var proxyAuth *proxy.Auth
 		socksURL, proxyErr := url.Parse(types.ProxySocksURL)
```

Comment:
we should check for error in proxyErr before proceeding with proxy creation or we'll run into nil pointer errors with invalid proxy urls